src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -300,7 +300,75 @@ def minimize_corpus(self, target_path, arguments, input_dirs, output_dir,
     Returns:
       A FuzzResult object.
     """
-    raise NotImplementedError
+    runner = _get_runner(target_path)
+
+    # Step 1: Generate corpus file for Centipede.
+    full_corpus_workdir = self._create_temp_dir('full_corpus_workdir')
+    input_dirs_param = ','.join(str(dir) for dir in input_dirs)
+    args = [
+        f'--workdir={full_corpus_workdir}',
+        f'--binary={target_path}',
+        f'--corpus_dir={input_dirs_param}',
+        '--num_runs=0',
+    ]
+    result = runner.run_and_wait(additional_args=args, timeout=max_time)
+    max_time -= result.time_executed
```

Comment:
Can you also add a check to make sure max_time > 0 and raise TimeoutError in all cases? We can just add this as an "and" to the existing `if result.timed_out` cases right?

Reply:
Done.
But am I correct that having `max_time < 0` after the subtraction implies the previous execution has timed out (`time_executed > max_time`), which is already captured by `if result.timed_out`?

Reply:
Ah, good point! But let's be defensive here just in case.